pkg/yqlib/datetime_preprocessor.go

```diff
@@ -0,0 +1,191 @@
+		return true
+	}
+
+	// Try to parse date-only format
+	if _, err := time.Parse("2006-01-02", value); err == nil {
+		return true
+	}
+
+	// Try RFC3339 without timezone
+	if _, err := time.Parse("2006-01-02T15:04:05", value); err == nil {
+		return true
+	}
+
+	return false
+}
+
+// PreprocessDocument automatically adds !!timestamp tags to ISO8601 datetime strings
+// in YAML documents when using Goccy parser for consistent datetime arithmetic behaviour
+func (dtp *DateTimePreprocessor) PreprocessDocument(yamlContent string) string {
+	if !dtp.enabled {
+		return yamlContent
+	}
+
+	lines := strings.Split(yamlContent, "\n")
+	var result strings.Builder
+
+	for i, line := range lines {
+		processed := dtp.processLine(line)
+		result.WriteString(processed)
+
+		// Add newline except for the last line (preserve original ending)
+		if i < len(lines)-1 {
+			result.WriteString("\n")
+		}
+	}
+
+	return result.String()
+}
+
+// processLine processes a single line of YAML, adding timestamp tags where appropriate
+func (dtp *DateTimePreprocessor) processLine(line string) string {
+	// Skip lines that are comments, already have tags, or are part of multi-line constructs
+	if isSkippableLine(line) {
+		return line
+	}
+
+	// Look for key-value pairs: "key: value" or "- value"
+	trimLeft := strings.TrimLeft(line, " \t")
+	if strings.HasPrefix(trimLeft, "- ") {
+		// Handle array items first (before key-value pairs)
+		return dtp.processArrayItemLine(line)
+	} else if colonIndex := strings.Index(line, ":"); colonIndex != -1 {
+		// Handle map key-value pairs
+		return dtp.processKeyValueLine(line, colonIndex)
+	}
+
+	return line
+}
+
+// isSkippableLine checks if a line should be skipped from datetime preprocessing
+func isSkippableLine(line string) bool {
+	trimmed := strings.TrimSpace(line)
+
+	// Skip empty lines, comments, directives, document separators
+	if trimmed == "" || strings.HasPrefix(trimmed, "#") ||
+		strings.HasPrefix(trimmed, "%") || strings.HasPrefix(trimmed, "---") ||
+		strings.HasPrefix(trimmed, "...") {
+		return true
+	}
+
+	// Skip lines that already have explicit tags
+	if strings.Contains(line, "!!") {
+		return true
+	}
+
+	// Skip multi-line scalar indicators
+	if strings.HasSuffix(trimmed, "|") || strings.HasSuffix(trimmed, ">") ||
+		strings.HasSuffix(trimmed, "|-") || strings.HasSuffix(trimmed, ">-") ||
+		strings.HasSuffix(trimmed, "|+") || strings.HasSuffix(trimmed, ">+") {
+		return true
+	}
+
+	return false
+}
+
+// processKeyValueLine processes a line containing a key-value pair
+func (dtp *DateTimePreprocessor) processKeyValueLine(line string, colonIndex int) string {
+	beforeColon := line[:colonIndex]
+	afterColon := line[colonIndex+1:]
+
+	// Check if the value part looks like a datetime
+	trimmedValue := strings.TrimSpace(afterColon)
+
+	// Skip if value is empty, quoted, or already complex
+	if trimmedValue == "" ||
+		strings.HasPrefix(trimmedValue, "\"") ||
```

Comment:
String literals notation can help

```suggestion
		strings.HasPrefix(trimmedValue, `"`) ||
```